scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -199,13 +210,35 @@ public int schedule() {
                     neededResourcesNumber = getNextcompatibleTasks(jobMap, taskRetrievedFromPolicy,
                             freeResourcesNb, tasksToSchedule);
                 }
+                if (logger.isDebugEnabled()) {
+                    logger.debug("tasksToSchedule : " + tasksToSchedule);
+                }
+
+
                 logger.debug("required number of nodes : " + neededResourcesNumber);
-                if (neededResourcesNumber == 0) {
+                if (neededResourcesNumber == 0 || tasksToSchedule.isEmpty()) {
                     break;
                 }
 
                 NodeSet nodeSet = getRMNodes(jobMap, neededResourcesNumber, tasksToSchedule);
 
+                if (nodeSet != null) {
```

Comment:
Could you wrap this logic into a separate private method with a meaningful name? It will help the readability